### DISCUSSION TITLE
AVA-type orientation maps

### DISCUSSION THREAD

**Discussion by ChrisTulley** [mtex-toolbox/mtex#749] [Ask Anything]
Hi team

I've been asked to make an AVA-type orientation map (grains or pixels colored according to the orientation of a specified crystal axis, in a sample reference frame), for some quartz ebsd data. I've been given some code that does that, but it only works for old mtex versions (currently running it on V5.03). I'd like to make one of these maps in a new version of mtex - can anyone help me to translate my code (pasted below) to work with version 5.6.0?

I've never made these sort of maps before, so I'm feeling my way in the dark here. I'm also a bit wary of what I'm doing - mtex is throwing me warnings about the colormap not being topologically correct. Any input much appreciated!

Chris Tulley

omA = TSLOrientationMapping(crystalSymmetry('-1'));
omA.colorPostRotation = rotation('axis',vector3d(0,1,0),'angle',180*degree)*rotation('axis',vector3d(0,0,1),'angle',-90*degree)*reflection(xvector);

figure(1) % plot color key
plot(omA)

% c-axis 
hs = Miller(0,0,0,1,ebsd('qtz').CS) 

figure(2) % plot orientation map
plot(grains,omA.Miller2color(grains.meanOrientation.*hs))
hold on
plot(grains.boundary)

**Comment by kilir** [comment]
Hi,
hava look at `HSVDirectionKey` with a `specimenSymmetry('-1')`. It will allow you to colorcode axes (discontinuously over the equator) in specimen coordinates.

Here's an example (hope it doesn't solve any homework assignments):
```
mtexdata titanium
plotzOutOfPlane
cs = ebsd('t').CS

% c-axis direction
c_dir = ebsd('t').orientations*Miller(0,0,0,1,cs);
% direction mapping - half sphere
dM = HSVDirectionKey(specimenSymmetry('-1'))

% compute rgb from direction
color = dM.direction2color(c_dir);

plot(ebsd('t'),color)

% some checks for visualization
cS= crystalShape.hex(cs);
grains = ebsd.calcGrains;
hold on
plot(grains('t').centroid +grains('t').meanOrientation.*cS.*50,'FaceAlpha',0.4)
hold off

nextAxis
plot(c_dir,color,'upper')
nextAxis
plot(dM)
```

Cheers,
Rüdiger

**Reply by ChrisTulley** [reply]
Thanks Rüdiger,

It seems to be working now, the plots show what I expect. Although, I still get  " Warning: Not a topological correct colormap! Please use the point group "1" " when I calcuate the color key with:

`dM = HSVDirectionKey(specimenSymmetry('-1'))`

I tried with specimenSymmetry('1'), and that also makes a sensible-looking plot. Is there a reason to either -1 or 1 for specimin symmetry?

PS. no homework to solve - just a PhD student who's been asked to turn some of my data into figures for teaching :)

Cheers,
Chris

**Reply by kilir** [reply]
Hi Chris,

let's assume you work on quartz which would mean you cannot distinguish between  +c and -c and the directions are axial data. Hence a half sphere is sufficient to color code the axes. 
However, our universe has the problem, that we can't have a unique color code without jumps between very close directions, usually at the equator - some axis which plunges with 0.1° to the east is very close to an axis which plunges with 0.1° to the west, however, there'll be a huge jump in color. (There's one grain in the titanium demo which shows this effect).

If you use the entire sphere, you won't have jumps (compare `plot(dM)`), however usually you can't decide which direction a c-axis points to.

Does that make sense?

Cheers,
Rüdiger

**Reply by ChrisTulley** [reply]
Thanks for your help Rüdiger, that all makes sense. 

Cheers,
Chris

**Reply by ChrisTulley** [reply]
Hi again @kilir 

Thanks again for your help in Feb - I've got another couple of questions:

I've plotted the map of [0001] orientations, and a lower hemisphere pole figure of the ebsd orientations within 3 selected grains. If I set 'plotzOutOfPlane', then the unit cell orientations are consistent with a lower hemisphere pole figure, but the color key for [0001] orientations needs to be rotated 180° about the page normal in order to show the directions on the lower hemisphere. Am I correct in thinking that the color key is set up to show directions on the upper hemisphere?

If I set 'plotzIntoPlane', then the unit cell orientations become consistent with the color key, but the lower hemisphere pole figure *appears* as an upper hemisphere plot. I'm not sure how to fix this inconsistency between the AVA maps, map color keys, and pole figures, other than by manually rotating the color key 180° about the page normal if I'm plotting with 'plotzOutOfPlane', or asking for the opposite hemisphere pole figure to the one I want to show, if I'm plotting with 'plotzIntoPlane'.  Is there a better option to make the AVA map, color key, and lower-hemisphere pole figure consistent?

Attached my code, incase that makes the issue clearer. 

`
mtexdata titanium

% plotting preference
%plotzOutOfPlane
plotzIntoPlane
plotx2east

cs = ebsd('t').CS

[grains,ebsd.grainId] = ebsd.calcGrains;

% c-axis direction
c_dir = ebsd('t').orientations*Miller(0,0,0,1,cs);

% direction mapping - half sphere
dM = HSVDirectionKey(specimenSymmetry('-1'))

% compute rgb from direction
color = dM.direction2color(c_dir);

% unit cell oriented as the grain mean
% orientation, scaled by grain size.
unitcell = grains('t').meanOrientation * crystalShape.hex(cs) * 0.35*(grains('t').grainSize);

% grains to mark
grain1 = grains(grains.id(40))
grain2 = grains(grains.id(36))
grain3 = grains(grains.id(42))

%% Plot
% color key for [0001] orientations
plot(dM)

% [0001] orientaion map and crystal shapes
nextAxis
plot(ebsd,color)
hold on
plot(grains.boundary)
hold on
plot(grains('t').centroid + unitcell)
hold on
plot(grain1.boundary,'linecolor','r','linewidth',3)
hold on
plot(grain2.boundary,'linecolor','g','linewidth',3)
hold on
plot(grain3.boundary,'linecolor','b','linewidth',3)

% pole figure
nextAxis
plotPDF(ebsd(grain1).orientations,Miller(0,0,0,1,cs,'uvtw'),'hemisphere','upper','markerColor','r')
hold on
plotPDF(ebsd(grain2).orientations,Miller(0,0,0,1,cs,'uvtw'),'markerColor','g')
hold on
plotPDF(ebsd(grain3).orientations,Miller(0,0,0,1,cs,'uvtw'),'markerColor','b')
`

**Comment by zmichels** [comment]
Hi,

On this page there is an example of changing (rotating or reflecting the color key). Perhaps some combination of these procedures might give you a matching color key?
Something to try at least.

https://mtex-toolbox.github.io/HomepageOld/files/doc/ColorCodingEBSD_demo

Cheers,
Zach

**Reply by ChrisTulley** [reply]
Hi Zach, 

Thanks for the suggestion - when using plotzOutOfPlane, applying a rotation to the color key after plotting the map would make the color key appear as it should for directions shown on the lower hemisphere. 

I'm still confused as to why there seems to be a mis-match between the unit cell orientations and the pole figure when using plotzIntoPlane, though.

Cheers,
Chris

**Comment by lfgmorales** [comment]
Hi Chris,



Are you using Aztec-Oxford?


If yes, you might be facing a know issue, please have a look here


https://www.jiscmail.ac.uk/cgi-bin/wa-jisc.exe?A2=ind1601&L=GEO-TECTONICS&P=R31836&K=2


This was also discusswed in the MTEX list sometime ago, but I don't find the url


you may also want to look at this paper:


https://www.sciencedirect.com/science/article/pii/S1044580316300924



cheers



Luiz





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
________________________________

**Reply by ChrisTulley** [reply]
Hi Luiz,

Thanks, yes, we do use AZtec-Oxford, but the apparant 'mis-matches' appear when I run the script using the titanium demo data (script pasted above), so I don't think it's related to the aquisition system. It's probably something i'm just not understanding, rather than a bug, but can't work out why the behaviour occurs. 

cheers,
Chris
